akka-slf4j/src/main/scala/akka/event/slf4j/Slf4jLogger.scala

```diff
@@ -138,7 +138,13 @@ class Slf4jLogger extends Actor with SLF4JLogging with RequiresMessageQueue[Logg
   protected def formatTimestamp(timestamp: Long): String =
     Helpers.currentTimeMillisToUTCString(timestamp)
 
-  private val actorSystemName = context.system.name
+  private val actorSystemName = context.system match {
+    case ex: ExtendedActorSystem =>
+      ex.provider.akkaSystem
+    case _ =>
+      context.system.name
+  }
```

Comment:
I'm not sure I like this reuse/change of `sourceActorSystem`. Sometimes being the system.name and sometimes being a full address. 

A better name would be `akkaAddress`, always formatted as an `Address`, but it's too late for 2.6.x to rename.

Changing the format might be ok, so we could have this as `sourceActorSystem` but always formatted as an `Address`.

Then we could also remove that part from `akkaSource`.

Then it's possible to concatenate `sourceActorSystem` and `akkaSource` to have a full path with address.

Reply:
I think it would be fine to not re-use but introduce a new `akkaAddress` in both classic and typed, given that `sourceActorSystem` never worked in typed yet, and then make that always have address format. 

Removing it from `akkaSource` I'm still not sure about, maybe that would be to likely to break some user logging setup?

Reply:
Plan:

* add `sourceActorSystem` to typed,  always actor system name (to align with classic)
* add `akkaAddress` which is always formatted as address
* drop system from `akkaSource` and only keeping the path

Reply:
`sourceThread` info is already present in the log events so trike that, also looked into dropping system from `akkaSource` and that is super invasive on the classic side so I changed my mind about that (would be weird to be inconsistent between typed/classic).